launcher/FileSystem.cpp

```diff
@@ -414,24 +414,29 @@ bool createShortCut(QString location, QString dest, QStringList args, QString na
 #endif
 }
 
-bool overrideFolder(QString overwritten_path, QString override_path)
+bool mergeFolders(QString dstpath, QString srcpath)
 {
-    using copy_opts = fs::copy_options;
-
-    if (!FS::ensureFolderPathExists(overwritten_path))
-        return false;
-
-    std::error_code err;
-    fs::copy_options opt = copy_opts::recursive | copy_opts::overwrite_existing;
-
-    fs::copy(toStdString(override_path), toStdString(overwritten_path), opt, err);
-
-    if (err) {
-        qCritical() << QString("Failed to apply override from %1 to %2").arg(override_path, overwritten_path);
-        qCritical() << "Reason:" << QString::fromStdString(err.message());
+    std::error_code ec;
+    fs::path fullSrcPath = srcpath.toStdString();
+    fs::path fullDstPath = dstpath.toStdString();
+    for (auto& entry : fs::recursive_directory_iterator(fullSrcPath))
+    {
+        fs::path relativeChild = fs::relative(entry, fullSrcPath);
+        if (entry.is_directory())
+            if (!fs::exists(fullDstPath / relativeChild))
+                fs::create_directory(fullDstPath / relativeChild);
+        if (entry.is_regular_file())
+        {
+            fs::path childDst = fullDstPath / relativeChild;
```

Comment:
why is this declared here instead of higher up

Reply:
it's declared inside the if's scope, since it's not used outside of that scope, why declare it anywhere else?